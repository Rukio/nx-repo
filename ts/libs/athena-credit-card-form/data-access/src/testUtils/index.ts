export * from './setupTestStore';
export * from './helpers';
