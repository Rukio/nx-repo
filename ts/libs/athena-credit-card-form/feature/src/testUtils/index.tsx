import { ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { configureStore, ConfigureStoreOptions } from '@reduxjs/toolkit';
import { authSlice } from '@*company-data-covered*/auth0/data-access';
import {
  athenaCreditCardFormApiSlice,
  collectPaymentSlice,
  creditCardSlice,
  RootState,
} from '@*company-data-covered*/athena-credit-card-form/data-access';
import { setupListeners } from '@reduxjs/toolkit/query';
import { Provider } from 'react-redux';
import type { Config as UserEventSetupConfig } from '@testing-library/user-event/setup/setup';
import { MemoryRouter, MemoryRouterProps } from 'react-router-dom';

type CustomRenderOptions = {
  renderOptions?: Omit<RenderOptions, 'queries'>;
  userEventOptions?: Omit<UserEventSetupConfig, 'delay'>;
  withRouter?: boolean;
  routerProps?: MemoryRouterProps;
  preloadedState?: ConfigureStoreOptions<RootState>['preloadedState'];
};

const customRender = (
  ui: ReactElement,
  options: CustomRenderOptions = {
    withRouter: true,
  }
) => {
  const {
    withRouter,
    routerProps = {},
    userEventOptions,
    renderOptions,
    preloadedState,
  } = options;

  const store = configureStore({
    reducer: {
      [authSlice.name]: authSlice.reducer,
      [athenaCreditCardFormApiSlice.reducerPath]:
        athenaCreditCardFormApiSlice.reducer,
      [collectPaymentSlice.name]: collectPaymentSlice.reducer,
      [creditCardSlice.name]: creditCardSlice.reducer,
    },
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware({
        immutableCheck: false,
        serializableCheck: false,
      }).concat(athenaCreditCardFormApiSlice.middleware),
    preloadedState,
  });

  setupListeners(store.dispatch);

  const component = withRouter ? (
    <MemoryRouter {...routerProps}>{ui}</MemoryRouter>
  ) : (
    ui
  );

  return {
    user: userEvent.setup({ ...userEventOptions, delay: null }),
    ...render(<Provider store={store}>{component}</Provider>, renderOptions),
  };
};

export * from '@testing-library/react';
export { customRender as render, userEvent };
