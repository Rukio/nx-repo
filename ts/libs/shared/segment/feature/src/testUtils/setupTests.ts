import '@*company-data-covered*/shared/testing/react/setup';
