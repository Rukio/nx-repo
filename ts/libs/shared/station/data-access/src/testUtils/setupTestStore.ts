import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { authSlice, AUTH_FEATURE_KEY } from '@*company-data-covered*/auth0/data-access';
import { stationApiSlice } from '../';

export const setupTestStore = () =>
  configureStore({
    reducer: combineReducers({
      [AUTH_FEATURE_KEY]: authSlice.reducer,
      [stationApiSlice.reducerPath]: stationApiSlice.reducer,
    }),
    middleware: (getDefaultMiddleware) =>
      getDefaultMiddleware().concat(stationApiSlice.middleware),
  });
