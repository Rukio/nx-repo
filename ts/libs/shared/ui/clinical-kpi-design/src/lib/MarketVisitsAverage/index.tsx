import {
  Typography,
  Box,
  makeSxStyles,
  useTheme,
} from '@*company-data-covered*/design-system';
import ProgressBar from '../ProgressBar';
import { FC } from 'react';
import MARKET_VISITS_AVERAGE_TEST_IDS from './testIds';

interface MarketVisitsAverageProps {
  /** The number of hours used to calculate market averages. */
  hoursWorked: number;
  userPatientCountAverage: number;
  marketPatientCountAverage: number;
}

const makeStyles = () =>
  makeSxStyles({
    root: {
      display: 'flex',
      flexDirection: 'column',
      alignItems: 'left',
      gap: 0.5,
    },
    hoursCalculatedText: (theme) => ({
      marginBottom: 1,
      color: theme.palette.text.secondary,
    }),
  });

const MarketVisitsAverage: FC<MarketVisitsAverageProps> = ({
  hoursWorked,
  userPatientCountAverage,
  marketPatientCountAverage,
}) => {
  const theme = useTheme();
  const styles = makeStyles();
  const yourProgressBarText = `You cared for ${userPatientCountAverage} patients`;
  const marketProgressBarText = `The Market average was ${marketPatientCountAverage} patients`;

  return (
    <Box
      sx={styles.root}
      data-testid={MARKET_VISITS_AVERAGE_TEST_IDS.MARKET_VISITS_AVERAGE_SECTION}
    >
      <Typography variant="subtitle2">
        Your Visits vs. Market Average
      </Typography>
      <Typography variant="body2" sx={styles.hoursCalculatedText}>
        The Market Average was calculated using {hoursWorked} hours worked.
      </Typography>
      <ProgressBar
        color={theme.palette.info.light}
        count={userPatientCountAverage}
        text={yourProgressBarText}
      />
      <ProgressBar
        color={theme.palette.secondary.dark}
        count={marketPatientCountAverage}
        text={marketProgressBarText}
      />
    </Box>
  );
};

export default MarketVisitsAverage;
