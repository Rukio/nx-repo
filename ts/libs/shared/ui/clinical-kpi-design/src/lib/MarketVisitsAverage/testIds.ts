const MARKET_VISITS_AVERAGE_TEST_IDS = {
  MARKET_VISITS_AVERAGE_SECTION: 'market-visits-average-section-test-id',
};

export default MARKET_VISITS_AVERAGE_TEST_IDS;
