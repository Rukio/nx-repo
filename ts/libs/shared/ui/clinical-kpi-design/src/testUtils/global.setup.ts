module.exports = async () => {
  process.env.TZ = process.env.TZ || 'America/Denver';
};
