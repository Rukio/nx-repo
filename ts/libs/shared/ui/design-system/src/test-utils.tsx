import React from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { ThemeProvider, theme } from './lib';
import type { Config as UserEventSetupConfig } from '@testing-library/user-event/setup/setup';
import userEvent from '@testing-library/user-event';

type CustomRenderOptions = {
  renderOptions?: Omit<RenderOptions, 'wrapper'>;
  userEventOptions?: UserEventSetupConfig;
};

const WrappedProviders: React.FC<{ children: React.ReactNode }> = ({
  children,
}) => {
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const customRender = (
  ui: React.ReactElement,
  options?: CustomRenderOptions
) => {
  return {
    user: userEvent.setup(options?.userEventOptions),
    ...render(ui, { wrapper: WrappedProviders, ...options?.renderOptions }),
  };
};

export * from '@testing-library/react';
export { customRender as render, userEvent };
