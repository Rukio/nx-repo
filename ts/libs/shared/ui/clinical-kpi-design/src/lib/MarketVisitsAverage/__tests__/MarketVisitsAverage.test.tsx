import { render, screen } from '../../../testUtils/test-utils';
import { BasicMarketVisitsAverage } from '../__storybook__/MarketVisitsAverage.stories';
import MARKET_VISITS_AVERAGE_TEST_IDS from '../testIds';

describe('Snapshot tests', () => {
  test('should render', () => {
    const { asFragment } = render(<BasicMarketVisitsAverage />);
    expect(asFragment()).toMatchSnapshot();
  });
});

describe('Unit tests', () => {
  test('should render correct content', () => {
    render(<BasicMarketVisitsAverage />);
    expect(
      screen.getByTestId(
        MARKET_VISITS_AVERAGE_TEST_IDS.MARKET_VISITS_AVERAGE_SECTION
      )
    ).toBeInTheDocument();
    expect(
      screen.getByText(
        'The Market Average was calculated using 8 hours worked.'
      )
    ).toBeInTheDocument();
    expect(screen.getByText('You cared for 7 patients')).toBeInTheDocument();
    expect(
      screen.getByText('The Market average was 5 patients')
    ).toBeInTheDocument();
  });
});
