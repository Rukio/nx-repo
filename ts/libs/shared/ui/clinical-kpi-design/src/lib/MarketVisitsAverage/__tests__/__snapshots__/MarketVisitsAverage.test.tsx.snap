// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot tests should render 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  gap: 4px;
}

.emotion-1 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 0.875rem;
  line-height: 1.57;
  letter-spacing: 0.00714em;
}

.emotion-2 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: rgba(0, 0, 0, 0.6);
  margin-bottom: 8px;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 4px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4 {
  box-sizing: border-box;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 2px;
}

.emotion-5 {
  width: 24px;
  height: 12px;
  border-radius: 4px;
  background-color: #03a9f4;
}

.emotion-12 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.75rem;
  line-height: 1.66;
  letter-spacing: 0.03333em;
  color: rgba(0, 0, 0, 0.87);
}

.emotion-15 {
  width: 24px;
  height: 12px;
  border-radius: 4px;
  background-color: #7b1fa2;
}

<div
    class="MuiBox-root emotion-0"
    data-testid="market-visits-average-section-test-id"
  >
    <h6
      class="MuiTypography-root MuiTypography-subtitle2 emotion-1"
    >
      Your Visits vs. Market Average
    </h6>
    <p
      class="MuiTypography-root MuiTypography-body2 emotion-2"
    >
      The Market Average was calculated using 8 hours worked.
    </p>
    <div
      class="MuiBox-root emotion-3"
    >
      <div
        class="MuiGrid-root emotion-4"
      >
        <div
          class="MuiBox-root emotion-5"
        />
        <div
          class="MuiBox-root emotion-5"
        />
        <div
          class="MuiBox-root emotion-5"
        />
        <div
          class="MuiBox-root emotion-5"
        />
        <div
          class="MuiBox-root emotion-5"
        />
        <div
          class="MuiBox-root emotion-5"
        />
        <div
          class="MuiBox-root emotion-5"
        />
      </div>
      <span
        class="MuiTypography-root MuiTypography-caption emotion-12"
      >
        You cared for 7 patients
      </span>
    </div>
    <div
      class="MuiBox-root emotion-3"
    >
      <div
        class="MuiGrid-root emotion-4"
      >
        <div
          class="MuiBox-root emotion-15"
        />
        <div
          class="MuiBox-root emotion-15"
        />
        <div
          class="MuiBox-root emotion-15"
        />
        <div
          class="MuiBox-root emotion-15"
        />
        <div
          class="MuiBox-root emotion-15"
        />
      </div>
      <span
        class="MuiTypography-root MuiTypography-caption emotion-12"
      >
        The Market average was 5 patients
      </span>
    </div>
  </div>
</DocumentFragment>
`;
